### PR TITLE
fixed the issue with the prep_107_vrf_lites.py

### LINKED ISSUE #256: TASK Prepare Service Model errors out if 'bgp' key is empty in the vrf_lite data model
### Ansible Version

```shell
ansible [core 2.16.3]
```

### Ansible Collection Versions

```shell
Collection         Version
------------------ -------
ansible.netcommon  5.3.0  
ansible.posix      1.5.4  
ansible.utils      2.11.0 
cisco.dcnm         3.6.0  
cisco.nac_dc_vxlan 0.3.0
cisco.nxos         5.3.0  
community.general  8.5.0
```

### Cisco NDFC Version

```shell
Version 12.2.1.316
```

### Cisco NX-OS Version

_No response_

### Which role is this issue related to?

cisco.nac_dc_vxlan.validate

### Which section of the data model is this issue related to?

vxlan.overlay_extensions

### Expected Behavior

Validation role should not throw a plugin error.

### Actual Behavior

```shell
TASK [cisco.nac_dc_vxlan.validate : Prepare Service Model] ********************************************************************************************
task path: /Users/devegupt/workspace/sac-ndfc/nac-ndfc/collections/ansible_collections/cisco/nac_dc_vxlan/roles/validate/tasks/sub_main.yml:100
Friday 13 December 2024  18:13:47 +0530 (0:00:00.020)       0:00:09.202 ******* 
Friday 13 December 2024  18:13:47 +0530 (0:00:00.020)       0:00:09.201 ******* 
[WARNING]: VRF-Lite in VXLAN as Code is currently not supported.
The full traceback is:
Traceback (most recent call last):
  File "/Users/devegupt/.pyenv/versions/3.10.13/envs/sac-ndfc/lib/python3.10/site-packages/ansible/executor/task_executor.py", line 165, in run
    res = self._execute()
  File "/Users/devegupt/.pyenv/versions/3.10.13/envs/sac-ndfc/lib/python3.10/site-packages/ansible/executor/task_executor.py", line 637, in _execute
    result = self._handler.run(task_vars=vars_copy)
  File "/Users/devegupt/workspace/sac-ndfc/nac-ndfc/collections/ansible_collections/cisco/nac_dc_vxlan/plugins/action/common/prepare_service_model.py", line 86, in run
    results=results).prepare()
  File "/Users/devegupt/workspace/sac-ndfc/nac-ndfc/collections/ansible_collections/cisco/nac_dc_vxlan/plugins/action/common/prepare_plugins/prep_107_vrf_lites.py", line 239, in prepare
    if af in vrf_lite["bgp"]:
TypeError: argument of type 'NoneType' is not iterable
fatal: [copy_netascode4_vrf_lite_ebgp -> localhost]: FAILED! => 
  msg: 'Unexpected failure during module execution: argument of type ''NoneType'' is not iterable'
  stdout: ''

NO MORE HOSTS LEFT ************************************************************************************************************************************

PLAY RECAP ********************************************************************************************************************************************
copy_netascode4_vrf_lite_ebgp : ok=19   changed=0    unreachable=0    failed=1    skipped=3    rescued=0    ignored=0   
```
```


### Ansible Playbook

```yaml
roles > validate > tasks > sub_main.yml


- name: Prepare Service Model
  cisco.nac_dc_vxlan.common.prepare_service_model:
    inventory_hostname: "{{ inventory_hostname }}"
    hostvars: "{{ hostvars }}"
    model_data: "{{ model_data['data'] }}"
    default_values: "{{ defaults }}"
    templates_path: "{{ role_path }}/../dtc/common/templates/"
  register: smd
  delegate_to: localhost
```

### Steps to Reproduce

```shell
Use the vrf_lite data model as below where "bgp" key is empty:


---
vxlan:
  overlay_extensions:
    vrf_lites:
      - name: myvrf_50001_vrf_lite
        vrf: myvrf_50001
        bgp:
          # local_as: 1111
          # best_path_as_path_relax: true
          # graceful_restart: true # Query: This doesn't show in the cli config.
          # graceful_restart_helper: true
          # address_family_ipv4_unicast:
          #     additional_paths_receive: true
          #     additional_paths_send: true
          #     additional_paths_selection_route_map: test-map-globalaf4
          #     default_originate: true
          #     ebgp_distance: 25
          #     ibgp_distance: 180
          #     local_distance: 150
          # address_family_ipv6_unicast:
          #     additional_paths_receive: true
          #     additional_paths_send: true
          #     additional_paths_selection_route_map: test-map-globalaf6
          #     default_originate: true
          #     ebgp_distance: 25
          #     ibgp_distance: 180
          #     local_distance: 200
        redistribution:
          - source: static
            # route_map: fabric-rmp-redist-static
        switches:
          - name: netascode4-ebgp-bl1
            router_id: 1.1.1.1
            loopback_id: 101
            bgp:
              local_as: 2222
              best_path_as_path_relax: true
              graceful_restart: true
              graceful_restart_helper: true
              address_family_ipv4_unicast:
                additional_paths_receive: true
                additional_paths_send: true
                additional_paths_selection_route_map: test-map-switchaf4
                default_originate: true
                ebgp_distance: 25
                ibgp_distance: 180
                local_distance: 100
              address_family_ipv6_unicast:
                additional_paths_receive: true
                additional_paths_send: true
                additional_paths_selection_route_map: test-map-switchaf6
                default_originate: true
                ebgp_distance: 25
                ibgp_distance: 180
                local_distance: 200
            static_routes:
              - prefix: 0.0.0.0/0
                route_tag: 12345
                next_hops:
                  - ip: 10.31.95.95
                  - ip: 10.31.96.96
              - prefix: 1.1.1.1/32
                route_tag: 12344
                next_hops:
                  - ip: 10.31.94.95
                  - ip: 10.31.93.96
            redistribution:
              - source: static
                # route_map: fabric-rmp-redist-static
            bgp_peers:
              - address: 10.31.95.95
                remote_as: 65099
                local_as: 3333
                description: BGP Peer 1
                update_source: Loopback101
                bfd:
                  enabled: true
                disable_connected_check: true
                remove_private_as: true
                remove_private_as_all: true
                ebgp_multihop: 2
                address_family_ipv4_unicast:
                    route_map_out: extcon-rmap-filter-v6
                    route_map_in: route-map-azure-in
                    weight: 100
                    allow_as_in: false
                    allow_as_in_number: 3
                    send_community: true
                    send_ext_community: true
                    as_override: true
                    next_hop_self: true
                    default_originate: true
                    route_reflector_client: false
                address_family_ipv6_unicast:
                    route_map_out: extcon-rmap-filter-v6
                    route_map_in: route-map-azure-in
                    weight: 100
                    allow_as_in: true
                    allow_as_in_number: 3
                    send_community: true
                    send_ext_community: true
                    as_override: true
                    next_hop_self: true
                    default_originate: true
                    route_reflector_client: false
                password: 014903c077689077e5b4548432c3d91d # 3DES encrypted password
```

### Relevant Debug Output

_No response_

### PR DESCRIPTION
**Pull request by devegupt**
## Related Issue(s)
Aims to resolve #256 

## Related Collection Role

* [x] cisco.nac_dc_vxlan.validate
* [ ] cisco.nac_dc_vxlan.dtc.create
* [ ] cisco.nac_dc_vxlan.dtc.deploy
* [ ] cisco.nac_dc_vxlan.dtc.remove
* [ ] other

## Related Data Model Element

* [ ] vxlan.global
* [ ] vxlan.topology
* [ ] vxlan.underlay
* [ ] vxlan.overlay_services
* [x] vxlan.overlay_extensions
* [ ] vxlan.policy
* [ ] defaults.vxlan
* [ ] other

## Proposed Changes

Fix the issue with the prep_107_vrf_lites.py plugin. Error is encountered if 'vxlan>overlay_extensions>vrf_lites>bgp' key is null.

## Cisco NDFC Version

12.2.1

## Checklist

* [x] Latest commit is rebased from develop with merge conflicts resolved
* [ ] New or updates to documentation has been made accordingly
* [x] Assigned the proper reviewers

### CODE REVIEW

**Comment by mikewiebe** [comment]
Is this ready for review?  If so please add the label.

File: plugins/action/common/prepare_plugins/prep_107_vrf_lites.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ def prepare(self):
                         #                   ebgp_distance: 25
                         #                   ibgp_distance: 180
                         #                   local_distance: 200
-                        if "bgp" in vrf_lite:
+                        if "bgp" in vrf_lite and vrf_lite["bgp"] is not None:
```
Comment (mikewiebe) [inline plugins/action/common/prepare_plugins/prep_107_vrf_lites.py:237]: Should be able to simply this to (but please test)

```python
if vrf_lite.get("bgp") is not None:
```

Comment (devegupt) [reply plugins/action/common/prepare_plugins/prep_107_vrf_lites.py]: I tested and the code you suggested works, I pushed the update. Thanks.

**Review by mikewiebe — APPROVED** [review]

File: plugins/action/common/prepare_plugins/prep_107_vrf_lites.py (hunk 1 above)
Comment (mikewiebe) [reply plugins/action/common/prepare_plugins/prep_107_vrf_lites.py]: Will merge as soon as the tests complete/pass

**Pull request merged**